llama_stack/providers/tests/agents/fixtures.py

```diff
@@ -60,9 +71,28 @@ async def agents_stack(request):
         if fixture.provider_data:
             provider_data.update(fixture.provider_data)
 
+    inf_provider_id = providers["inference"][0].provider_id
+    safety_provider_id = providers["safety"][0].provider_id
+
+    shield = get_shield_to_register(
+        providers["safety"][0].provider_type, safety_provider_id, safety_model
+    )
+
+    inference_models = (
+        inference_model if isinstance(inference_model, list) else [inference_model]
+    )
     impls = await resolve_impls_for_test_v2(
         [Api.agents, Api.inference, Api.safety, Api.memory],
         providers,
         provider_data,
+        models=[
+            Model(
+                identifier=model,
```

Comment:
This is super annoying by the way when it shows up in the config. There are two things which are "terrible" which I plan to fix. The fixes might be annoying too.

- First, the names "identifier" and "provider_resource_id" must be used and the not the aliases `model_id`, `provider_model_id` which show up in the `register_model()` API
- As structured now, `provider_resource_id` and `provider_id` cannot be optional either. I think at the very least `provider_resource_id` might need to be optional.

Reply:
See #429